src/cmd/ksh93/tests/variables.sh

```diff
@@ -507,13 +511,12 @@ fi
 { $SHELL -c '
 function foo
 {
-	typeset SECONDS=0
-	sleep .002
+	SECONDS=0.0
+	sleep 0.002
 	print $SECONDS
-
 }
 x=$(foo)
-(( x >.001 && x < 1 ))
+(( x >= 0.002 && x < 1 ))
```

Comment:
If you are tightening up the lower bounds, why not have it be x > 0.002? if x==.002 then something is wrong as the print command should consume some time cycles as well.

Reply:
Good catch on the first part. I encountered test failures on Mac OS X where it came in exactly on the mark for 0.002.

Reply:
Isn't SECONDS always defined on the Korn shell?

Reply:
That makes sense as I am not sure the granularity of time cycles that can be caught by the SECONDS variable.

Reply:
Yes, SECONDS is defined by ksh at startup but it can be deleted. That typeset constrains the altering of the SECONDS variable to zero to be local to only function foo instead of altering the global scope.
```ksh
$ function x { typeset SECONDS=0; sleep .2; print $SECONDS; }; echo $SECONDS; x; echo $SECONDS
1745.464
0.205
1745.669
$ function x { SECONDS=0; sleep .2; print $SECONDS; }; echo $SECONDS; x; echo $SECONDS
1762.015
0.200
0.200
$
```